src/storage/src/source/postgres.rs

```diff
@@ -349,25 +349,39 @@ async fn ensure_replication_slot(client: &Client, slot: &str) -> Result<(), Tran
     }
 }
 
+/// The state of a replication slot.
+struct SlotMetadata {
+    /// The process ID of the session using this slot if the slot is currently actively being used.
+    /// None if inactive.
+    active_pid: Option<i32>,
```

Comment:
I'm confused about this! I was thinking we'd emit the backend PID for _our_ connection, but this is emitting the backend PID, if any, for the connection that was using the slot previously? I _think_ that tends to already get printed to the logs when the "replication slot active for PID" error gets hit later when we try to claim the replication slot.

Reply:
Ugh, tokio postgres makes this hard to get at. Working on it here: https://github.com/MaterializeInc/rust-postgres/pull/24